tests/lib/muzic/src/integration.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <zlib.h>
+
+const char sample[]
+    = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do "
+      " eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, "
+      " quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo conse quat.  "
+      "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat"
+      " nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui    "
+      "officia deserunt mollit anim id est laborum.";
+
+z_stream* deflater;
+z_stream* inflater;
+
+static void setup() {
+  deflater = (z_stream*)malloc(sizeof(z_stream));
+  zassert_not_equal(deflater, NULL, NULL);
+  deflater->next_in = Z_NULL;
+  deflater->avail_in = 0;
+  deflater->total_in = 0;
+  deflater->total_out = 0;
+  deflater->zalloc = Z_NULL;
+  deflater->zfree = Z_NULL;
+  deflater->opaque = Z_NULL;
+  int res = deflateInit(deflater, Z_DEFAULT_COMPRESSION);
+  zassert_equal(res, Z_OK, "deflate initialization failed: %d", res);
+
+  inflater = (z_stream*)malloc(sizeof(z_stream));
+  zassert_not_equal(inflater, NULL, NULL);
+  inflater->next_in = Z_NULL;
+  inflater->avail_in = 0;
+  inflater->total_in = 0;
+  inflater->total_out = 0;
+  inflater->zalloc = Z_NULL;
+  inflater->zfree = Z_NULL;
+  inflater->opaque = Z_NULL;
+  res = inflateInit(inflater);
+  zassert_equal(res, Z_OK, "inflate initialization failed: %d", res);
+}
+
+static void teardown() {
+  deflateEnd(deflater);
+  free(deflater);
+  deflater = NULL;
+  inflateEnd(inflater);
+  free(inflater);
+  inflater = NULL;
+}
+
+static void test_deflate_finish() {
+  unsigned char defl_buf[1024];
```

Comment:
Might not be great to allocate such a large amount on the stack.

Please make static / shared among all test cases.